tests/test_integration.py

```diff
@@ -714,13 +714,20 @@ async def test_hoprd_check_ticket_price_is_default(peer, swarm7):
 @pytest.mark.parametrize("src,dest", random_distinct_pairs_from(default_nodes(), count=PARAMETERIZED_SAMPLE_SIZE))
 async def test_peeking_messages_with_timestamp(src, dest, swarm7):
     message_count = int(TICKET_AGGREGATION_THRESHOLD / 10)
+    split_index = int(message_count * 0.66)
+
     random_tag = random.randint(10, 65530)
 
     src_peer = swarm7[src]
     dest_peer = swarm7[dest]
 
     packets = [f"0 hop message #{i:08d}" for i in range(message_count)]
-    for packet in packets:
+    for packet in packets[:split_index]:
+        await src_peer["api"].send_message(dest_peer["peer_id"], packet, [], random_tag)
+
+    await asyncio.sleep(2)
```

Comment:
Please, replace this unnecessary sleep with an explicit wait on the other side (peek all packets).

Reply:
Isn't the problem here that you should sleep after each packet, not just after all of them are sent?

Reply:
In real world, not while testing, we will get packet generated with the same timestamp. IMO waiting between each packet will not solve the issue.
This 2s sleep is to ensure to have different timestamp so that I can peek only the most recents message (otherwise all message would have the same timestamp -> would be retrieved by peek-all)